### DISCUSSION TITLE
Which EXIF tags are mapped to which PhotoPrism fields?

### DISCUSSION THREAD

**Discussion by djjudas21** [photoprism/photoprism#3351] [General]
I'm writing a script to add EXIF tags into JPG files that don't have them, because they are scanned from film negatives.  I have some questions about which EXIF tags are mapped onto which PhotoPrism fields.

![Screenshot from 2023-04-26 15-17-07](https://user-images.githubusercontent.com/604595/234606326-84d3606e-612e-4cc9-9e1a-6fc470cbd57f.png)

In this example, which EXIF fields correspond to `Title`, `Description` and `Notes`?

I am considering writing tags to set not only the title & caption of the image, but also a block of text to describe which film stock was used, how it was developed, etc.

Thanks.

**Comment by graciousgrey** [comment]
You can find the mapping here: https://www.photoprism.app/kb/metadata :)

**Comment by djjudas21** [comment]
Amazing, thanks @graciousgrey

**Comment by djjudas21** [comment]
Actually, are you sure this is correct?

Your [table of metadata](https://www.photoprism.app/kb/metadata) says that the PhotoPrism field `Title` maps to the EXIF tag `Title`, but I can't see any such tag. The [exiftool docs](https://exiftool.org/TagNames/EXIF.html) list `XPTitle` but it seems that `ImageDescription` is actually the real field for the title.

There's also no `Description` or `Comment` tags in exiftool. The nearest matches are `ImageDescription`, `UserComment` and `XPComment`.

**Reply by lastzero** [reply]
Title is an XMP field, Dublin Core to be precise. Exif does not have a Title field, so most software uses Description there.

**Reply by djjudas21** [reply]
So if I write an EXIF tag to `ImageDescription`, PhotoPrism will display it as the title?

**Reply by graciousgrey** [reply]
The table refers to Exiftool tags. Those include https://exiftool.org/TagNames/EXIF.html and https://exiftool.org/TagNames/XMP.html. You can write to those fields using 

`exiftool -Title="Your title" -Comment="Your Notes" image.jpg`

**Comment by djjudas21** [comment]
The library I'm using for writing these tags is pyexiv2 and that seems aligned with exiftool: https://exiv2.org/tags.html
